### PR TITLE
✨ amp-next-page: Send document title/URL changes to the viewer

### PR DESCRIPTION
**Pull request by peterjosling**
Implements the draft spec for the updated viewer history API.

Sends document title, path and canonical URLs to the viewer on document change.

### CODE REVIEW

**Comment by peterjosling** [comment]
@aghassemi

File: src/service/history-impl.js [hunk 1]
```diff
@@ -542,8 +542,10 @@ export class HistoryBindingNatural_ {
     return this.whenReady_(() => {
       const newState = this.mergeStateUpdate_(
           this.getState_(), opt_stateUpdate || {});
-      this.historyReplaceState_(newState, /* title */ undefined,
-          newState.fragment ? ('#' + newState.fragment) : undefined);
+      const url = (newState.url || '').replace(/#.*/, '');
```
Comment (aghassemi) [inline src/service/history-impl.js]: We also need to update the `replace_` for the viewer case. But I can take care of that. For now we need to protect the extra params to be send to the viewer.

For now please wrap the existing implt in https://github.com/ampproject/amphtml/blob/3d331aba53737f2464707dcff90ba644c107b01a/src/service/history-impl.js#L907 in a `if (!opt_stateUpdate.url)` and add a TODO for me for the else case. What's involved is checking the capability of the viewer and sending a different type of message.

File: test/unit/test-history.js [hunk 2]
```diff
@@ -528,6 +528,28 @@ describes.sandboxed('HistoryBindingNatural', {}, () => {
       expect(onStateUpdated.getCall(3).args[0].stackIndex).to.equal(h);
     });
   });
+
+  it('should update path from URL parameter', () => {
```
[the remark this answers is not in the document]
Comment (peterjosling) [reply test/unit/test-history.js]: Title isn't used in the native history APIs. Do we want the AMP API to set document.title automatically?

File: src/service/history-impl.js (hunk 1 above)
Comment (peterjosling) [reply src/service/history-impl.js]: The viewer case is already sending the full opt_stateUpdate object in the message, is that an issue?

Comment (peterjosling) [reply src/service/history-impl.js]: Gotcha, done.

**Comment by aghassemi** [comment]
There are some legit unit test failures here. Let's investigate

**Comment by aghassemi** [comment]
@peterjosling All the failing tests manipulate a global history :( working on it

**Pull request merged**